src/components/InputField/InputField.tsx

```diff
@@ -0,0 +1,87 @@
+import { COLORS } from "@src/constants/colors";
+import { useEffect, useState } from "react";
+import { TextInput, View, TouchableOpacity } from "react-native";
+
+import { styles, inputStyles } from "./InputField.styles";
+import { InputFieldProps } from "./InputField.types";
+import { ErrorIcon } from "./assets/ErrorIcon";
+import { PasswordHideIcon } from "./assets/PasswordHideIcon";
+import { PasswordShowIcon } from "./assets/PasswordShowIcon";
+import { ValidIcon } from "./assets/ValidIcon";
+import { Typography } from "../Typography";
+
+export const InputField = ({
+  initVal = "",
+  getVal,
+  errorConditions = false,
+  errorChildren = "Textfield is invalid",
+  password = false
+}: InputFieldProps) => {
+  const [textVal, setTextVal] = useState(initVal);
```

Comment:
```suggestion
  const [value, setValue] = useState<string>(initVal);
```
our type can tell us if the state is string or not. how do you think?

Reply:
Any reason to keep this state here?

Reply:
no reason. thank you for let me know!!

Reply:
Let's remove this then!! :) also please do not forget to remove `useEffect` hook as well.